move gold storage to separate repo

tomeichlersmith opened the pull request:
This moves the gold histogram and log files into their own repository [LDMX-Software/ci-data](https://github.com/LDMX-Software/ci-data) so that normal human developers do not need to wait for the large files to be downloaded whenever they `git clone` or `git pull`.

The ci-data repository will *not* have any commit history. I did this in order to save space since the gold histogram files are binary and do not benefit from git's delta algorithm. Each time we make a new release, the generate_pr_gold workflow will forcibly amend the last commit on ci-data to the new gold, keeping the repo's size (and number of commits) constant.

> [!CAUTION]
> This setup allows for *anyone* on *any branch* to write a GitHub workflow in ldmx-sw that makes arbitrary commits to the LDMX-Software/ci-data repository using the CI_DATA_PAT token.

I don't think this is something we need to worry about though because if someone does it by accident, we will ask them to stop and if someone does it on purpose we can remove their access from the LDMX-Software/ldmx-sw repo. In either case, we can re-push the correct gold files by either creating a new release again, re-running a prior release's action, or manually doing it locally.

Not very many people try to mess with the github workflows so I don't expect this to happen, but I do think its important for the ldmx-sw admins to know about the possibility.

Using the ci-data repository is pretty easy, we just `git clone` it if we need the gold files - opting in to downloading them. Since the workflow updates the commit by amend, you should not `git pull` within the ci-data repository. Instead, just remove the repo and re-clone.

tomeichlersmith committed:
delete gold from ldmx-sw to make sure we use copies from other repo

tomeichlersmith committed:
attempt to use ci-data repo

tvami commented:
Looks like it's working!

Did you think about https://github.com/LDMX-Software/ldmx-sw/issues/1694#issuecomment-2802268006 ? 
I think this should be possible as a force push to trunk

tomeichlersmith committed:
fix path to gold log

tomeichlersmith committed:
draft non-writeable update to gold gen

tomeichlersmith committed:
update label file and write commit message

use new token, rename job, and delete ldmx-sw checkout

tomeichlersmith commented:
Alright, I think I got writing working. I am going to enable `on: push` in the generate pr golden histograms and then check `ci-data` to make sure it got updated.

tomeichlersmith committed:
patches to generate_pr_gold for it to function

- remove common.sh script source and usage of its commands
- update paths
- separate groups into steps
- fix typo to commit.txt
- update commit message

tomeichlersmith commented:
Alright, I've gotten the following error a few different ways, so I think I need to return to the security settings.

```
 [main f6bac52] ci data for release refs/heads/1694-gold-storage\n tag corresponds to commit 35a945f0fc208fe94a807b0a01a596cf2e7b3335 of LDMX-Software/ldmx-sw
 15 files changed, 11071 insertions(+), 11071 deletions(-)
remote: Permission to LDMX-Software/ci-data.git denied to tomeichlersmith.
fatal: unable to access 'https://github.com/LDMX-Software/ci-data/': The requested URL returned error: 403
Error: Process completed with exit code 128.
```

tomeichlersmith commented:
Alright, I used some other dummy repositories to figure out the token stuff and so I think I got that figured out. If this works, then I'll create one more token with the same permissions but without an expiration date.

Try 8 (update commit msg, add `--amend` back): https://github.com/LDMX-Software/ldmx-sw/actions/runs/15472657579?pr=1721

tomeichlersmith committed:
update workflow notes

bryngemark reviewed (approved):
Looks sensible, glad we can save some space and time here. Do you think this needs any more discussion or is SWAN advertisement upon next release sufficient? (The latter is my inclination.)
